### PR TITLE
Feat: Homepage, Darkmode

### PR DESCRIPTION
**Pull request by yashpatel1797**
Change Log
- Homepage
- Category section
- dark mode

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -0,0 +1,32 @@
+.App{
+    min-height: calc(100vh - 5rem);
+    /* background-color: var(--color-grey-light-1); */
```
Comment (rohanmathur91) [inline src/App.css:3]: remove commented code

File: src/App.css [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+}
+.dark-theme .text-intro,
+.dark-theme .login-form {
+  color: #a0aec0;
+  background-color: #2b2f3d;
+}
+.dark-theme .nav-bar {
+  color: #a0aec0;
+  background-color: #212726;
+  background-color: #434047;
```
Comment (rohanmathur91) [inline src/App.css:13]: try to use color variables, also there are two background colors given to the `.nav-bar` remove one

**Review by rohanmathur91 — APPROVED** [review]
- Code is understandable
- Good that you have broken the page into several components
- Suggestion: Run prettier in CSS file as there is no line space between classes

File: jsconfig.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+    "compilerOptions": {
+        "baseUrl": "src"
+    },
+    "include": ["src"]
+}
```
Comment (ravikumar0403) [inline jsconfig.json:6]: Good that you have used path aliasing

File: src/components/Card/Card.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+                    alt={data.categoryName}
+                    className="card-image"
+                />
+            </div>
+            <div className="text-container">
+                <div className="text-container-title">
+                    <p>{data.categoryName}</p>
+                </div>
+                <div className="text-container-desc">
+                    <p>Take this quiz to test your knowledge on {data.categoryName}!</p>
```
Comment (ravikumar0403) [inline src/components/Card/Card.js:19]: You can destructure data

File: src/config/routes.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+const routes = [
+    {
+        path: "/",
+        element: <Home />,
+    },
+    {
+        path: "/mockMan",
+        element: <MockMan />,
+    },
+]
```
Comment (ravikumar0403) [inline src/config/routes.js:12]: Good way to defining all routes in a separate file.

File: src/context/theme-context.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    const darkThemeBtn = () => {
+        document.body.classList.add("dark-theme");
+        localStorage.setItem("theme", "dark");
+        setTheme("dark");
+    }
+    const lightThemeBtn = () => {
+        document.body.classList.remove("dark-theme");
+        localStorage.setItem("theme", "light");
+        setTheme("light")
+    }
```
Comment (ravikumar0403) [inline src/context/theme-context.js:25]: These functions have same logic so you can create single function as pass the variable theme as a parameter

**Review by ravikumar0403 — APPROVED** [review]
Home.module.css is empty file, you can remove it if not required,
I've left few comments apart from that everything looks good

File: src/context/theme-context.js (hunk 6 above)
Comment (yashpatel1797) [reply src/context/theme-context.js:25]: sure

**Pull request merged**